### PR TITLE
GN-4714: improvements to html-editor

### PR DESCRIPTION
**Pull request by elpoelma**
### Overview
This PR contains improvements to the html-editor contained in the `debug-tools` component:
- Removal of obsolete/unused xml-editor features
- Addition of a reusable `codemirror` modifier with support for custom extensions
   * codemirror-editor can also be update 'from the outside': if any of the modifier params change (`content`, `onUpdate` or `extensions`), the codemirror editor is correctly updated, without causing infinite re-validation/loop bugs.
- Improvements to html formatting
   * Replaced `xml-formatter` by `js-beautify`
   * Content located in the following tags is not formatted: `p`, header-tags, `span`, `a`
- Introduction of seperate html-editor modal component
- Addition of toolbar-button which allows opening the html-editor
- Adjustments to styles of toolbar-button component to better support text children

TODO:

- [x] Add seperate component which opens up the `html-editor` modal with improved UI
- [x] (optional) Improve `code-mirror` modifier so it reacts to content changes from the outside

##### connected issues and PRs:
[GN-4714](https://binnenland.atlassian.net/browse/GN-4714?atlOrigin=eyJpIjoiZGVjNDBhZTMwZTE0NDYxOWFjY2RmNmQ4ZWRlNzA3ODUiLCJwIjoiaiJ9)

### How to test/reproduce
- `pnpm start`
- insert any sample data
- open up the html editor
- notice the html is formatted, except for phrasing content between the aforementioned tags
- when setting the html, the spacing should be unchanged
- notice that the `xml` editor option is gone

### Challenges/uncertainties
While working on this feature, I once again noticed a lot of inconsistencies in the styles of our toolbar components. We should probably tackle these in a seperate PR.

### Checks PR readiness
- [x] UI: works on smaller screen sizes
- [x] UI: feedback for any loading/error states
- [x] Check cancel/go-back flows
- [x] Check database state correct when deleting/updating (especially regarding relationships)
- [x] changelog
- [x] npm lint
- [x] no new deprecations

### CODE REVIEW

File: addon/components/plugins/html-editor/modal.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  onCancel: () => void;
+};
+
+export default class HTMLEditorModal extends Component<Args> {
+  CodeMirror = CodeMirrorModifier;
+
+  @localCopy('args.content') declare content: string;
+
+  get formattedContent() {
+    return beautify.html(this.content, {
```
Comment (piemonkey) [inline addon/components/plugins/html-editor/modal.ts:21]: Looking at some examples in the editor, I find it easier to read (though a little aggressively spread onto new lines) with e.g. the ``wrap_attributes:` 'force-expand-multiline'` option too.

File: tests/dummy/app/components/sample-toolbar-responsive.hbs [hunk 2]
```diff
@@ -46,7 +46,14 @@
   <:side as |Tb|>
```
Comment (piemonkey) [inline tests/dummy/app/components/sample-toolbar-responsive.hbs:46]: Hmm, not necessarily something for now, but this side menu is getting crowded and isn't handling fitting content well... 
image

**Review by piemonkey — COMMENTED** [review]
Looks good but I'm seeing unexpected behaviour.

- If I set some content using edit HTML, (say `<p>hello world<p>`), then change it in the 'real' editor (e.g. change hello -> hi), then refresh the browser, I get the HTML edit version rather than the modified one (`<p>hello world<p>` instead of `<p>hi world<p>`.

- I get strange behaviour on newlines. For example, this is just pressing 'enter' on an empty `<p></p>`: 
Peek 2024-03-06 15-29

- It would be amazing to override the default tab behaviour to avoid it jumping to the 'save' button...

File: tests/dummy/app/components/sample-toolbar-responsive.hbs (hunk 2 above)
Comment (elpoelma) [reply tests/dummy/app/components/sample-toolbar-responsive.hbs:46]: Agreed, we should probably tackle this as part of this ticket: https://binnenland.atlassian.net/browse/GN-4737?atlOrigin=eyJpIjoiNGFlNmNkMzRlY2JmNDcxNWE4NGQyMGNmMTdlNDFhMjgiLCJwIjoiaiJ9 (redesign of the toolbar)

**Comment by elpoelma** [comment]
Thanks for the comments!

* The strange behaviour when inserting newlines has been fixed in https://github.com/lblod/ember-rdfa-editor/pull/1151/commits/35957b5e320fc7631e2434f99eece908bcb2eb7f
* I added a commit to override the default browser tab behaviour in https://github.com/lblod/ember-rdfa-editor/pull/1151/commits/9f0d8a0d01c41989917d081b4221df6d6ad6d714

File: addon/components/plugins/html-editor/modal.ts (hunk 1 above)
Comment (elpoelma) [reply addon/components/plugins/html-editor/modal.ts]: Good point, I'll also look into the other possible values of `wrap_attributes`

**Comment by elpoelma** [comment]
> If I set some content using edit HTML, (say <p>hello world<p>), then change it in the 'real' editor (e.g. change hello -> hi), then refresh the browser, I get the HTML edit version rather than the modified one (<p>hello world<p> instead of <p>hi world<p>.

Fixed in https://github.com/lblod/ember-rdfa-editor/pull/1151/commits/1e8597f57d35c65c3ec3a36a79bfb29afb46ca6d :facepalm:

**Review by piemonkey — APPROVED** [review]
Nice! I kicked off the pipelines.

**Pull request merged**